.github/mergify.yml

```diff
@@ -15,7 +15,7 @@ pull_request_rules:
     # Check redhat-distro-container job only when relevant paths are changed
     - or:
       - and:
-        - check-success=build-test-push
+        - check-success~=build-test-push
```

Comment:
without knowing the exact semantics, I'm always +1 to use flexible expression except than (sub)string matches. You could still check that there are word separators in front and after the word to avoid matches on (hypothethical) `no-build-test-push` which could even be introduced in the future. In Perl RE syntax this would be `\bbuild-test-push\b`, but I'm not sure if this is honoured here (alternatively, just check for a trailing and leading space with `\sbuild-test-push\s` or quote and use spaces literally.

Reply:
Of course you could check for `^build-test-push$`, too if you want an *exact* string match (`^` for the beginning of a line, `$` for the end of a line, see also the matches below that leverages those anchors)